packages/ui-view/src/View/styles.js

```diff
@@ -0,0 +1,569 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import { DIRECTION } from '@instructure/ui-i18n'
+import {
+  getShorthandPropValue,
+  mirrorShorthandEdges,
+  mirrorShorthandCorners
+} from '@instructure/ui-themeable'
+import { pickProps } from '@instructure/ui-react-utils'
+import { error } from '@instructure/console/macro'
+import generateComponentTheme from './theme'
+
+const getBorderStyle = ({ borderRadius, borderWidth, dir, theme }) => {
+  const isRtlDirection = dir === DIRECTION.rtl
+
+  return {
+    borderRadius: getShorthandPropValue(
+      'View',
+      theme,
+      isRtlDirection ? mirrorShorthandCorners(borderRadius) : borderRadius,
+      'borderRadius'
+    ),
+    borderWidth: getShorthandPropValue(
+      'View',
+
+      theme,
+      isRtlDirection ? mirrorShorthandEdges(borderWidth) : borderWidth,
+      'borderWidth'
+    )
+  }
+}
+
+const getSpacingStyle = ({ margin, padding, dir, theme }) => {
+  const isRtlDirection = dir === DIRECTION.rtl
+
+  return {
+    margin: getShorthandPropValue(
+      'View',
+      theme,
+      isRtlDirection ? mirrorShorthandEdges(margin) : margin,
+      'margin'
+    ),
+    padding: getShorthandPropValue(
+      'View',
+      theme,
+      isRtlDirection ? mirrorShorthandEdges(padding) : padding,
+      'padding'
+    )
+  }
+}
+
+const getOffsetStyle = ({
+  insetBlockStart,
+  insetBlockEnd,
+  insetInlineStart,
+  insetInlineEnd,
+  dir
+}) => {
+  const isRtlDirection = dir === DIRECTION.rtl
+
+  const blockStart = {
+    top: insetBlockStart,
+    insetBlockStart
+  }
+
+  const blockEnd = {
+    bottom: insetBlockEnd,
+    insetBlockEnd
+  }
+
+  const horizontalOffsets = {
+    left: isRtlDirection ? insetInlineEnd : insetInlineStart,
+    right: isRtlDirection ? insetInlineStart : insetInlineEnd,
+    insetInlineStart,
+    insetInlineEnd
+  }
+
+  return {
+    ...blockStart,
+    ...blockEnd,
+    ...horizontalOffsets
+  }
+}
+
+const getStyleProps = ({ cursor, style }) => {
+  const whitelisted = pickProps(style || {}, {}, [
+    // Position/calculateElementPosition:
+    'top',
+    'left',
+    'position',
+    'display',
+    'transform',
+    'overflow',
+    'minWidth',
+    'minHeight',
+    // Img:
+    'filter',
+    // Flex.Item:
```

Comment:
This syntax will be deprecated in #172, Flex.Item => FlexItem

Reply:
Fixed